lib/modules/manager/pipenv/artifacts.ts

```diff
@@ -113,35 +114,95 @@ export function getPipenvConstraint(
   return '';
 }
 
-function getMatchingHostRule(url: string): HostRule {
-  return find({ hostType: PypiDatasource.id, url });
+function getMatchingHostRule(url: string): HostRule | undefined {
+  const parsedUrl = parseUrl(url);
+  if (parsedUrl) {
+    parsedUrl.username = '';
+    parsedUrl.password = '';
+    const urlWithoutCredentials = parsedUrl.toString();
+
+    return find({ hostType: PypiDatasource.id, url: urlWithoutCredentials });
+  }
```

Comment:
```suggestion
  }
  return undefined;
```
missing return?

Reply:
omitted intentionally. i didnt want to introduce a reason for the CI to demand coverage for an explicit statement. But lets see if it does...